PrivateHospitals/backend/PrivateHospitals/PrivateHospitals.Infrastructure/Repositories/Appointment/AppointmentRepository.cs

```diff
@@ -18,25 +18,75 @@ public async Task<bool> CreateAppointmentAsync(Core.Models.Appointment appointme
+            .Take(appointmentFilter.PageSize)
+            .Select(x => new Core.Models.Appointment
+            {
+                AppointmentDate = x.AppointmentDate,
+                Doctor = new Core.Models.Users.Doctor()
+                {
+                    FirstName = x.Doctor.FirstName,
+                    LastName = x.Doctor.LastName
+                },
+                Patient = new Core.Models.Users.Patient()
+                {
+                    FirstName = x.Patient.FirstName,
+                    LastName = x.Patient.LastName
+                },
+                Status = x.Status
+            })
             .ToListAsync();
 
-        return appointments;
+        var result = new PaginationResult<Core.Models.Appointment>(items, totalItems, appointmentFilter.PageNumber,
+            appointmentFilter.PageSize);
+
+        return result;
     }
 
-    public async Task<List<Core.Models.Appointment>> GetAppointmentsByDateAsync(string patientId, DateTime? fromDate, DateTime? toDate)
-    {
-        var appointments = await _context.Appointments
-            .Include(x => x.Patient)
-            .Include(x => x.Doctor)
-            .Where(x => x.AppointmentDate >= fromDate && x.AppointmentDate <= toDate && x.PatientId == patientId)
-            .ToListAsync();
+    public async Task<PaginationResult<Core.Models.Appointment>> GetAppointmentsByDateAsync(AppointmentFilter appointmentFilter, string patientId)
+    { 
+        var totalItems = await _context.Appointments
+            .AsNoTracking()
+            .Where(x => x.AppointmentDate <= appointmentFilter.FromDate && x.AppointmentDate >= appointmentFilter.ToDate && x.PatientId == patientId)
+            .CountAsync();
+        
+        var items = await _context.Appointments
+            .AsNoTracking()
+            .Where(x => x.AppointmentDate >= appointmentFilter.FromDate && x.AppointmentDate <= appointmentFilter.ToDate && x.PatientId == patientId)
+            .OrderBy(x => x.AppointmentDate)
+            .Skip((appointmentFilter.PageNumber - 1) * appointmentFilter.PageSize)
+            .Take(appointmentFilter.PageSize)
+            .Select(x => new Core.Models.Appointment
+            {
+                AppointmentDate = x.AppointmentDate,
+                Doctor = new Core.Models.Users.Doctor()
+                {
+                    FirstName = x.Doctor.FirstName,
+                    LastName = x.Doctor.LastName
+                },
+                Patient = new Core.Models.Users.Patient()
+                {
+                    FirstName = x.Patient.FirstName,
+                    LastName = x.Patient.LastName
+                },
+                Status = x.Status
+            }).ToListAsync();
+        
+        var result = new PaginationResult<Core.Models.Appointment>(items, totalItems, appointmentFilter.PageNumber,
+            appointmentFilter.PageSize);
 
-        return appointments;
+        return result;
```

Comment:
Better `GetAppointmentsByFilter` and do the approach I mentioned with IQueryable